src/chipeitlib/src/main/java/es/chipeit/lib/core/InstructionSet.kt

```diff
@@ -46,42 +48,167 @@ internal fun ldVxByte(instruction: Int, registers: IRegisters) {
+    val y = instruction shr 1 * 4 and 0xF
+
+    var vx = registers.v[x].toInt() and 0xFF
+    var vy = registers.v[y].toInt() and 0xFF
+
+    val add = vx + vy
+
+    registers.v[0xF] = (add shr 8).toByte()
+    registers.v[x] = add.toByte()
+
+    registers.pc += 2
+}
 
 // 8xy5 - SUB Vx, Vy
+internal fun subVxVy(instruction: Int, registers: IRegisters) {
+    val x = instruction shr 2 * 4 and 0xF
+    val y = instruction shr 1 * 4 and 0xF
+
+    var vx = registers.v[x].toInt() and 0xFF
+    var vy = registers.v[y].toInt() and 0xFF
+
+    val add = vx - vy
+
+    registers.v[x] = add.toByte()
+    registers.v[0xF] = if (add > 0) 1 else 0
+
+    registers.pc += 2
+}
 
 // 8xy6 - SHR Vx {, Vy}
 
 // 8xy7 - SUBN Vx, Vy
+internal fun subnVxVy(instruction: Int, registers: IRegisters) {
+    val x = instruction shr 2 * 4 and 0xF
+    val y = instruction shr 1 * 4 and 0xF
+
+    var vx = registers.v[x].toInt() and 0xFF
+    var vy = registers.v[y].toInt() and 0xFF
+
+    val add = vy - vx
+
+    registers.v[0xF] = if (add > 0) 1 else 0
+
+    registers.v[x] = add.toByte()
+
+    registers.pc += 2
+}
 
 // 8xyE - SHL Vx {, Vy}
 
 // 9xy0 - SNE Vx, Vy
+internal fun sneVxVy(instruction: Int, registers: IRegisters) {
+    val x = instruction shr 2 * 4 and 0xF
+    val y = instruction shr 1 * 4 and 0xF
+
+    if (registers.v[x] != registers.v[y])
+        registers.pc += 2
+
+    registers.pc += 2
+}
 
 // Annn - LD I, addr
 
 // Bnnn - JP V0, addr
 
 // Cxkk - RND Vx, byte
 
-// Dxyn - RND Vx, byte
+// Dxyn - DRW Vx, Vy, Nibble
+internal fun drwVxVyNibble(
+        instruction: Int,
+        registers: IRegisters,
+        memory: IMemory<Byte>,
+        graphicMemory: ICoreGraphicMemory
+) {
+    val x = instruction shr 2 * 4 and 0xF
+    val y = instruction shr 1 * 4 and 0xF
+    val height = instruction and 0xF
+
+    val vx = registers.v[x].toInt() and 0xFF
+    val vy = registers.v[y].toInt() and 0xFF
+
+    var pixelCleared = false
+    for (i in 0 until height) {
+        val spriteRow = memory[registers.i + i]
+
+        if (graphicMemory.drawRow(vx, vy + i, spriteRow))
+            pixelCleared = true
+    }
+    registers.v[0xF] = if (pixelCleared) 1 else 0
+
+    registers.pc += 2
+}
 
 // Ex9E - SKP Vx
 
 // ExA1 - SKNP Vx
 
 // Fx07 - LD Vx, DT
+internal fun ldVxTimer(instruction: Int, registers: IRegisters, timer: ITimer) {
+    val x = instruction shr 2 * 4 and 0xF
 
-// Fx0A - LD Vx, K
+    registers.v[x] = timer.t
+
+    registers.pc += 2
+}
 
-// Fx15 - LD DR, Vx
+// Fx0A - LD Vx, K
 
+// Fx15 - LD DT, Vx
 // Fx18 - LD ST, Vx
+internal fun ldTimerVx(instruction: Int, registers: IRegisters, timer: ITimer) {
+    val x = instruction shr 2 * 4 and 0xF
+
+    timer.t = registers.v[x]
+
+    registers.pc += 2
+}
 
 // Fx1E - ADD I, Vx
 
 // Fx29 - LD F, Vx
+internal fun ldFVx(instruction: Int, registers: IRegisters) {
+    val x = instruction shr 2 * 4 and 0xF
+
+    var vx = registers.v[x].toInt() and 0xFF
+
+    if (vx > 0xF)
```

Comment:
Do not check this. It is not our dutty to check the correctness of the program. Plus, if the program tries to access a register over `0xF`, an exception will be raised on the memory, so there is no need to shadow that exception with this one.